### DISCUSSION TITLE
[Crash] 3.1.2 causes crash on the Nintendo Switch loading screen

### DISCUSSION THREAD

**Discussion by cherryssbu** [Raytwo/ARCropolis#291] [Issues] [answered]
### Have you read the disclaimer before proceeding?

- [X] I have read the disclaimer.

### System Firmware Version

14.0.0

### Atmosphere Version

1.3.1

### Smash Version

13.0.1

### ARCropolis Version

3.1.2

### Version channel

Stable

### SD card formatting

FAT32

### Can the crash be consistently reproduced?

Yes

### At what moment does the crash happen?

- [ ] Before the Switch boot animation
- [X] After the Switch boot animation
- [ ] During the Smash ball loading on boot
- [ ] Before the opening cutscene, but only sometimes
- [ ] On the Stage Select Screen
- [ ] On the Character Select Screen
- [ ] During the loading for a battle
- [ ] At random
- [ ] After a long time playing

### Context

I installed project HDR a smash mod onto my switch and noticed once I wanted to use just vanilla smash with some cosmetic mods I encountered this issue for the first time. To trouble shoot I removed and uninstalled any cosmetic mods, removed the training room mod, and uninstalled all files related to HDR and its plugins. I still had this issue so I was suggested to try uninstalling ARCropolis. Apon doing so and booting up smash again I was able to boot into the game and use the training mod as well. I then reinstalled ARCropolis and tried to run it alongside the training room mod and it then crashed (this time my ultimate/mods folder had no mods within it.) 

### Configuration

wat

### Crash report

AtmosphÃ¨re Crash Report (v1.7):
Result:                          0x4A8 (2168-0002)

Process Info:
    Process Name:                Application
    Program ID:                  01006a800016e000
    Process ID:                  0000000000000085
    Process Flags:               00000077
    User Exception Address:      00000041be551000
Exception Info:
    Type:                        Data Abort
    Address:                     0000000000000018
    Fault Address:               0000000000000000
Crashed Thread Info:
    Thread ID:                   0000000000000240
    Thread Name:                 Thread_0x00000029E369E670
    Stack Region:                0000002e9de89000-0000002e9e089000
    Registers:
        X[00]:                   0000000000177a02
        X[01]:                   0000000000000000
        X[02]:                   0000004f82500000
        X[03]:                   0000000000000000
        X[04]:                   00000020c74b33ac
        X[05]:                   0000000000000000
        X[06]:                   0000004f82500000
        X[07]:                   00000000000002a9
        X[08]:                   0000004f807a12d0
        X[09]:                   0000000000000000
        X[10]:                   0000000000000001
        X[11]:                   00000000fffffffd
        X[12]:                   0000000000000000
        X[13]:                   000000000000000a
        X[14]:                   0000000000000000
        X[15]:                   0000000000000000
        X[16]:                   00000029e3db46d0
        X[17]:                   00000029e3377fb0 (nnSdk + 0x1f2fb0) (_ZN2nn2fs6detail12FileAccessor5WriteElPKvmRKNS0_11WriteOptionE + 0x0)
        X[18]:                   0000000000000000
        X[19]:                   0000004f807b4bc0
        X[20]:                   00000020c74b33ac
        X[21]:                   0000004f807b4bc0
        X[22]:                   00000020c74b33ac
        X[23]:                   0000004f82500000
        X[24]:                   0000000000000000
        X[25]:                   0000000000000001
        X[26]:                   0000002e9e088dd1
        X[27]:                   0000000000000000
        X[28]:                   0000000000000000
        FP:                      0000002e9e088cc0
        LR:                      00000029e337aafc (nnSdk + 0x1f5afc) (_ZN2nn2fs9WriteFileENS0_10FileHandleElPKvmRKNS0_11WriteOptionE + 0xdc)
        SP:                      0000002e9e088ca0
        PC:                      00000029e3378058 (nnSdk + 0x1f3058) (_ZN2nn2fs6detail12FileAccessor5WriteElPKvmRKNS0_11WriteOptionE + 0xa8)
    Stack Trace:
        ReturnAddress[00]:       00000029e337aafc (nnSdk + 0x1f5afc) (_ZN2nn2fs9WriteFileENS0_10FileHandleElPKvmRKNS0_11WriteOptionE + 0xdc)
        ReturnAddress[01]:       00000020c7477f54 (arcropolis + 0x161f54)
        ReturnAddress[02]:       00000029e34b94b8 (nnSdk + 0x3344b8)
        ReturnAddress[03]:       00000029e34bd984 (nnSdk + 0x338984)
        ReturnAddress[04]:       0000000000000000
    Stack Dump:                               00 01 02 03 04 05 06 07 08 09 0a 0b 0c 0d 0e 0f
                                 002e9e088ca0 00 00 00 00 01 00 00 00 00 3f 75 80 00 00 00 00
                                 002e9e088cb0 00 00 00 00 00 00 00 00 a9 02 00 00 00 00 00 00
                                 002e9e088cc0 10 8d 08 9e 2e 00 00 00 fc aa 37 e3 29 00 00 00
                                 002e9e088cd0 20 8d 08 9e 2e 00 00 00 74 0f f3 de 29 00 00 00
                                 002e9e088ce0 00 00 00 00 00 00 00 00 43 58 c7 20 00 00 00 00
                                 002e9e088cf0 e0 1f 5a c7 20 00 00 00 d8 1f 5a c7 20 00 00 00
                                 002e9e088d00 a9 02 00 00 00 00 00 00 c8 8d 08 9e 2e 00 00 00
                                 002e9e088d10 80 8f 08 9e 2e 00 00 00 54 7f 47 c7 20 00 00 00
                                 002e9e088d20 50 8d 08 9e 2e 00 00 00 08 a8 28 80 4f 00 00 00
                                 002e9e088d30 08 39 35 c7 20 00 00 00 c8 a9 28 80 4f 00 00 00
                                 002e9e088d40 08 a8 28 80 4f 00 00 00 20 3f 75 80 4f 00 00 00
                                 002e9e088d50 80 8f 08 9e 2e 00 00 00 5c 39 35 c7 20 00 00 00
                                 002e9e088d60 63 72 6f 70 6f 6c 69 73 2f 63 6f 6e 66 69 67 2f
                                 002e9e088d70 73 64 3a 2f 75 6c 74 69 6d 61 74 65 2f 61 72 63
                                 002e9e088d80 20 3f 75 80 4f 00 00 00 8c 00 00 00 00 00 00 00
                                 002e9e088d90 38 f6 34 c7 20 00 00 00 00 00 00 00 00 00 00 00
    TLS Address:                 0000004051dc4400
    TLS Dump:                                 00 01 02 03 04 05 06 07 08 09 0a 0b 0c 0d 0e 0f
                                 004051dc4400 06 00 00 00 08 00 00 00 00 00 00 00 00 00 00 00
                                 004051dc4410 02 00 00 00 8b 00 00 00 00 00 00 00 00 00 00 00
                                 004051dc4420 53 46 43 4f 00 00 00 00 00 00 00 00 a1 6d 71 f3
                                 004051dc4430 00 00 00 00 00 00 00 00 00 00 00 00 00 00 00 00
                                 004051dc4440 00 00 00 00 00 00 00 00 00 00 00 00 00 00 00 00
                                 004051dc4450 00 00 00 00 00 00 00 00 00 00 00 00 00 00 00 00
                                 004051dc4460 00 00 00 00 00 00 00 00 00 00 00 00 00 00 00 00
                                 004051dc4470 00 00 00 00 00 00 00 00 00 00 00 00 00 00 00 00
                                 004051dc4480 00 00 00 00 00 00 00 00 00 00 00 00 00 00 00 00
                                 004051dc4490 00 00 00 00 00 00 00 00 00 00 00 00 00 00 00 00
                                 004051dc44a0 00 00 00 00 00 00 00 00 00 00 00 00 00 00 00 00
                                 004051dc44b0 00 00 00 00 00 00 00 00 00 00 00 00 00 00 00 00
                                 004051dc44c0 00 00 00 00 00 00 00 00 00 00 00 00 00 00 00 00
                                 004051dc44d0 00 00 00 00 00 00 00 00 00 00 00 00 00 00 00 00
                                 004051dc44e0 00 00 00 00 00 00 00 00 00 00 00 00 00 00 00 00
                                 004051dc44f0 00 00 00 00 00 00 00 00 00 00 00 00 00 00 00 00
Module Info:
    Number of Modules:           8
    Module 00:
        Address:                 00000029db5dc000-00000029db600000
        Name:                    [00000000]
        Module Id:               0000000000000000000000000000000000000000000000000000000000000000
    Module 01:
        Address:                 00000029db600000-00000029db602000... (9 KB left)

### Minimal modpack to reproduce the crash

It was legit just using ARCropolis

**Comment by maxface14** [comment]
I'm experiencing the same issue. Nothing but ARCropolis is enabled yet the game immediately crashes upon selecting the game and choosing a profile. Everything that can be up to date is up to date both on the jailbreak and the ultimate plugins.

**Comment by cherryssbu** [comment, marked as answer]
Idk what changed as of today but I was just attempting to see if the newest
version of arcropolis worked and everything on my end works fine. Not sure
what changed tbh.

**Comment by je19921** [comment]
I also have the same issue. tried completely removing my mods folder and only keeping arcopolis, and the game still crash.

**Reply by gucciboy512** [reply]
boot into hekate and try the "fix archive bits" or something like that. like @Raytwo mentioned, it is probably because of bad sectors on your msd card

**Reply by je19921** [reply]
Thank you, the fix archive bits function solved the issue. I reinstalled Arcopolis and skyline afterward and everything works flawlessly.

**Comment by Raytwo** [comment]
Hey OP,

Seems like ARCropolis is having some trouble writing to your SD. Maybe run a check on your PC for corrupted sectors?
Also lol at the "Configuration" field of your issue.

**Comment by DShip91** [comment]
You aren't trying to use the **Press Start to End Results Screen mod**, are you? That mod has been causing a lot of issues with Ultimate crashing on boot.

**Reply by Raytwo** [reply]
FWIW the author is aware, so maybe keep an eye on it.
